odxtools/ecuvariantmatcher.py

```diff
@@ -1,171 +1,32 @@
 # SPDX-License-Identifier: MIT
-from enum import Enum
-from typing import Dict, Generator, List, Optional
+from typing import List
 
-from .diaglayers.diaglayer import DiagLayer
-from .diaglayers.diaglayertype import DiagLayerType
 from .diaglayers.ecuvariant import EcuVariant
-from .diagservice import DiagService
-from .exceptions import OdxError, odxassert, odxrequire
-from .matchingparameter import MatchingParameter
-from .odxtypes import ParameterValue
-from .response import Response
+from .variantmatcher import VariantMatcher
 
 
-class EcuVariantMatcher:
-    """EcuVariantMatcher implements the matching algorithm of ecu variants according to their
-    ECU-VARIANT-PATTERN according to ISO 22901-1.
+class EcuVariantMatcher(VariantMatcher):
+    """EcuVariantMatcher implements the matching algorithm of ecu variants using
+    ECU-VARIANT-PATTERNs according to ISO 22901-1.
 
     Usage (example):
 
     ```python
 
-    # initialize the matcher with a list of ecu variants,
-    # i.e., DiagLayer instances of variant_type == DiagLayerType.ECU-VARIANT
+    # initialize the matcher with a list of ecu variants
     matcher = EcuVariantMatcher(ecu_variant_candidates=[...], use_cache=use_cache)
 
     # run the request loop to obtain responses for every request
-    for req in matcher.request_loop():
-        resp = ... # make a UDS request
+    for _, encoded_request in matcher.request_loop():
```

Comment:
Why return something if it's going to be discarded?

Reply:
because this allows to use the same code for ECU variant matching and base variant matching: for whatever reason, functional addressing can only be specified for the latter (oversight of the spec?)